php/razorpay_js/index.html

```diff
@@ -0,0 +1,402 @@
+			);
+			try {
+				let myHeaders = new Headers();
+				myHeaders.append("Content-Type", "application/x-www-form-urlencoded");
+
+				let urlencoded = new URLSearchParams();
+				urlencoded.append(
+					"subscription_items[item_price_id][0]",
+					ITEM_PRICE_ID
+				);
+				urlencoded.append("subscription_items[billing_cycles][0]", 2);
+				urlencoded.append("subscription_items[quantity][0]", 1);
+				urlencoded.append("cb_customer_id", CB_CUSTOMER_ID);
+
+				let requestOptions = {
+					method: "POST",
+					headers: myHeaders,
+					body: urlencoded,
+				};
+
+				let estimate = await fetch(
+					"http://localhost:3000/api/create_estimate",
+					requestOptions
+				)
+					.then((response) => response.json())
+					.then((result) => {
+						console.log(result);
+						return result;
+					})
+					.catch((error) => console.log("error", error));
+				return estimate;
+			} catch (error) {
+				console.log("error ========", error);
+			}
+		}
+
+		async function createRazorpayCustomer(customerData) {
+			console.info(
+				`======= Creating New Customer in Razorpay ========`
+			);
+			try {
+				let myHeaders = new Headers();
+				myHeaders.append("Content-Type", "application/json");
+
+				let urlencoded = {
+					name: customerData.name || "Test User 14",
+					contact: customerData.contact || "9123456782",
+					email: customerData.email || "testuser14@example.com"
+				};
+
+				let requestOptions = {
+					method: "POST",
+					headers: myHeaders,
+					body: JSON.stringify(urlencoded),
+				};
+
+				let newRazorpayCustomer = await fetch(
+					"http://localhost:3000/api/create_razorpay_customer",
+					requestOptions
+				)
+					.then((response) => response.json())
+					.then((result) => {
+						console.log(result);
+						return result;
+					})
+					.catch((error) => console.log("error", error));
+				return newRazorpayCustomer;
+			} catch (error) {
+				console.log("error ========", error);
+			}
+		}
+
+		async function createOrder(customerId, estimatedAmount, currencyCode) {
+			console.info(
+				`====== Creating Order in Razorpay =======`
+			);
+			try {
+				let myHeaders = new Headers();
+				// myHeaders.append("Authorization", RP_AUTH_HEADER);
+				myHeaders.append("Content-Type", "application/json");
+
+				let urlencoded = {
+					amount: estimatedAmount,
+					currency: currencyCode,
+					payment_capture: "0",
+					receipt: "",  // add receipt details if any
+					customer_id: customerId,
+				};
+
+				if(paymentType === 'upi') {
+					urlencoded.method = "upi";
+					urlencoded.token = {
+						"max_amount": 300000,  // declare your max amount
+						"expire_at": 2709971120,
+						"frequency" : "monthly"
+					}
+				}
+
+				let requestOptions = {
+					method: "POST",
+					headers: myHeaders,
+					body: JSON.stringify(urlencoded)
+				};
+
+				// console.log("url encoded", urlencoded, requestOptions)
+
+				let orderDetails = await fetch(
+					"http://localhost:3000/api/create_razorpay_order",
```

Comment:
Extract the server URL out to a global variable, then build the URLs while making the request